pkg/config/ao.go

```diff
@@ -294,20 +296,23 @@ func (ao *AOConfig) replaceAO(data []byte) error {
 	return nil
 }
 
-func (ao *AOConfig) getUpdateURL() string {
-	var updateCluster string
+func (ao *AOConfig) getUpdateURL() (string, error) {
 	for _, c := range ao.AvailableUpdateClusters {
 		available, found := ao.Clusters[c]
 		logrus.WithField("exists", found).Info("update server", c)
-		if found && available.Reachable {
-			updateCluster = c
-			break
+
+		if !found || (found && !available.Reachable) {
+			continue
+		}
+
+		serviceURLs, err := ao.GetServiceURLs(c)
```

Comment:
Hvorfor er det nødvendig å kjøre hele ao.GetServiceURLs() i en metode som kun skal finne updateURL for ao?

Kunne være fint med en test der denne feiler og man får neste alternativ fra AvailableUpdateClusters.

Reply:
Den støtter både gammel og ny konfigurasjon. Var lettest å bruke den, fremfor å lage en egen som kun forholder seg til en url. Hvordan ville du ha løst det?

Jeg kan fikse en test som gjør det du sier.

Reply:
Jeg ser jo at den gamel løsningen gjorde dette: return fmt.Sprintf(ao.UpdateURLPattern, updateCluster)

Jeg tror jeg ville laget en mer spisset metode som hentet updateURL basert på UpdateURLPattern for det aktuelle clusteret. GetServiceURLs() gjør veldig mye mer, noe som i verste fall kan hindre oppdatering når vi vil ha ut en bugfiks.

Reply:
Det blir en del copy/paste for å opprette den funksjonen. Men har laget et forslag til det.
https://github.com/Skatteetaten/ao/pull/187

Reply:
Jeg har svart på #187 og liker den bedre, men ser at det er noe duplisert kode rundt håndtering av clusterConfig.Type og Henting av ServiceUrlPatterns uansett hvilken løsning man velger.